airbyte-integrations/connectors/source-posthog/source_posthog/streams.py

```diff
@@ -104,9 +103,10 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         return {self.cursor_field: max(latest_state, current_state)}
 
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        self._initial_state = self._initial_state or stream_state.get(self.cursor_field) or self._start_date
```

Comment:
it seems like it would be better to set this variable in `read_records` instead something like: 

```
self._initial_state = ...
yield from super().read_records(**kwargs)
```
wdyt?